tests/integration/update_cluster/spotblock/in-v1alpha2.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: kops.k8s.io/v1alpha2
+kind: Cluster
+metadata:
+  creationTimestamp: "2016-12-10T22:42:27Z"
+  name: spotblock.example.com
+spec:
+  kubernetesApiAccess:
+    - 0.0.0.0/0
+  channel: stable
+  cloudProvider: aws
+  configBase: memfs://clusters.example.com/spotblock.example.com
+  etcdClusters:
+    - etcdMembers:
+        - instanceGroup: master-us-test-1a
+          name: us-test-1a
+        - instanceGroup: master-us-test-1b
+          name: us-test-1b
+        - instanceGroup: master-us-test-1c
+          name: us-test-1c
+      name: main
+    - etcdMembers:
+        - instanceGroup: master-us-test-1a
+          name: us-test-1a
+        - instanceGroup: master-us-test-1b
+          name: us-test-1b
+        - instanceGroup: master-us-test-1c
+          name: us-test-1c
+      name: events
+  kubernetesVersion: v1.12.9
+  masterInternalName: api.internal.spotblock.example.com
+  masterPublicName: api.spotblock.example.com
+  networkCIDR: 10.0.0.0/16
+  networking:
+    kubenet: {}
+  nonMasqueradeCIDR: 100.64.0.0/10
+  sshAccess:
+    - 0.0.0.0/0
+  topology:
+    masters: public
+    nodes: public
+  subnets:
+    - cidr: 10.0.1.0/24
+      name: us-test-1a
+      type: Public
+      zone: us-test-1a
+    - cidr: 10.0.2.0/24
+      name: us-test-1b
+      type: Public
+      zone: us-test-1b
+    - cidr: 10.0.3.0/24
+      name: us-test-1c
+      type: Public
+      zone: us-test-1c
+
+---
+
+apiVersion: kops.k8s.io/v1alpha2
+kind: InstanceGroup
+metadata:
+  creationTimestamp: "2016-12-10T22:42:28Z"
+  name: nodes
+  labels:
+    kops.k8s.io/cluster: spotblock.example.com
+spec:
+  associatePublicIp: true
+  image: kope.io/k8s-1.4-debian-jessie-amd64-hvm-ebs-2016-10-21
+  machineType: t3.medium
+  maxSize: 2
+  minSize: 2
+  role: Node
+  instanceProtection: true
+  maxPrice: "0.1"
+  spotDurationInMinutes: 120
```

Comment:
Apologies for the runaround but we're doing some significant refactoring in the integration tests (see #8737) and the timing of that and some other PRs is going to cause a merge conflict here.

I think it would be simpler to add this field to the existing `launch_templates` test cluster rather than creating a new integration test cluster. You could delete `tests/integration/update_cluster/spotblock` and revert your changes in `cmd/kops/integration_test.go`, then add the new field to `tests/integration/update_cluster/launch_templates/in-v1alpha2.yaml` and rerun `./hack/update-expected.sh` once more. It will produce a much smaller PR and wont conflict with the other changes we're making.

I hope this makes sense, let me know if you have any questions.

Reply:
I see that #8737 is not yet merged, will the above thing work if I rebase this branch with master?

Reply:
@rifelpet meant that would be easier for him to rebase and modify #8737 if you add this instance group to the existing `launch_templates` test instead of creating a new one.

Reply:
Sorry for my late reply. I've been a bit busy over the last few days.

Reply:
There's few complication in adding the integration test as part of `launch_templates`. The `TestLaunchTemplatesASG` fails since the method `runTestAWS` is made for one type of node, now when I add a new asg with spot-node, the actual is correct but the expected is mis formed because of `runTestAWS` method.

Should I refactor the `runTestAWS` to run for multiple nodes for single kube-cluster?

Reply:
Also there's an assumption that the instance group name is always `nodes` 
https://github.com/kubernetes/kops/blob/master/cmd/kops/integration_test.go#L486

Reply:
In this case, could you just add these to the existing IG for `launch_templates`?
```
  maxPrice: "0.1"
  spotDurationInMinutes: 120
```

Reply:
Seems like a quick fix, but will do.